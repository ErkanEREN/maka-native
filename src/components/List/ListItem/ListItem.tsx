import React from "react";
import { ListItem, ListItemProps } from "react-native-elements";

import styled from "styled-components";

interface Props {
  ListItemProps?: ListItemProps;
  item?: { name: string };
  style?: {};
}

export default function ListItemComponent({
  item,
  style,
  ListItemProps
}: Props) {
  return <ListItem {...ListItemProps} />;
}
