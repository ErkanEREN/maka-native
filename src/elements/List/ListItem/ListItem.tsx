import React from "react";
import { Text } from "react-native";

import styled from "styled-components";

interface Props {
  item?: {};
  style?: {};
}

export default function ListItemComponent({ item, style }: Props) {
  return <Text>{JSON.stringify(item)}</Text>;
}
