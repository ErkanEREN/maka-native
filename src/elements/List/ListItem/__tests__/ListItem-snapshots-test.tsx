import React from "react";
import renderer from "react-test-renderer";

import ListItem from "../ListItem";

describe("ListItem", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const item = {
        key: 1,
        value: "1"
      };
      const tree = renderer.create(<ListItem item={item} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
