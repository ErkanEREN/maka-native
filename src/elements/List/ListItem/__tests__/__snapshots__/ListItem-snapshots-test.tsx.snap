// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Rendering should match to snapshot 1`] = `
<Text>
  {"key":1,"value":"1"}
</Text>
`;
